build_library/build_image_util.sh

```diff
@@ -607,6 +607,25 @@ finish_image() {
        "${root_fs_dir}/boot/flatcar/vmlinuz-a"
   sudo rm "${root_fs_dir}/usr/boot/vmlinuz"*
 
+  # Forbid dynamic user ID allocation because we want stable IDs
+  local found=""
+  # We want to forbid "-", "X:-", "-:X", "-:-"
+  found=$({ grep '^[^#]' "${root_fs_dir}"/usr/lib/sysusers.d/*.conf || true ; } | awk '{print $3}' | { grep -x -- "-"$'\n'".*:-"$'\n'"-:.*" || true ; })
```

Comment:
I'd limit that to `u` and `g` entries (for users and groups) so:
```suggestion
  found=$({ grep '^[ug]' "${root_fs_dir}"/usr/lib/sysusers.d/*.conf || true ; } | awk '{print $3}' | { grep -x -- "-"$'\n'".*:-"$'\n'"-:.*" || true ; })
```

Reply:
About `"-"$'\n'".*:-"$'\n'"-:.*"`. I suppose that newlines are separating three patterns: `-`, `.*:-` and `-:.*`. So the second pattern will match the unlikely line like `/file/owned/by/user/:-`.

I'd change the pattern to `'-\|-:-\|[0-9]\+:-\|-:[0-9]\+'` (basically use `\|` instead of `$'\n'`, and `[0-9]\+` instead of `.*`, and add `-:-` as another subpattern).

Reply:
`/file/owned/by/user/:-` should be matched, too, because it means to pick a random group ID, or?

Reply:
> `/file/owned/by/user/:-` should be matched, too, because it means to pick a random group ID, or?

No, if it's an absolute path then it's treated as a whole as an absolute path, and from that path UID and GID are taken. But there is another corner case I haven't covered yet, probably. You could also use `uid:groupname` syntax, so I'd say that pattern `-:.*` was correct, whereas `-:[0-9]\+` is too narrow. For more info, there's `man sysusers.d` for describing the file format.

Reply:
You could also skip the grep and use `awk '{if ($3 !~ /^\// && ($3 ~ /^-:|:-$|^-$/)) {print $3}}'`, probably.

Reply:
Question: should we also be bailing out on paths, actually?

Reply:
> Question: should we also be bailing out on paths, actually?

Ok, let's do it for clarity

Reply:
> (I had thought `/path:-` would be valid, that's why I had asked)

It is valid, but it means "take owner UID and GID from file `path:-` located in `/`.